File: com-mini-core/src/main/java/com/mini/core/security/digest/SHA256.java
package com.mini.core.security.digest;

import java.nio.charset.Charset;

public final class SHA256 extends BaseDigest {
	public SHA256() {
		super("SHA-256");
	}
	
	public static SHA256 getInstance() {
		return new SHA256();
	}
	
	public static String encode(String data, Charset charset) {
		return getInstance().update(data, charset).encode();
	}
	
	public static String encode(String data) {
		return getInstance().update(data).encode();
	}
}

File: mini-util/src/main/kotlin/com/mini/core/security/digest/SHA256.kt
@file:Suppress("unused")

package com.mini.core.security.digest

import java.nio.charset.Charset

class SHA256 : BaseDigest("SHA-256") {
    companion object{

        val instance: SHA256
            get() = SHA256()

        fun encode(data: String?, charset: Charset?): String {
            return instance.update(data!!, charset!!).encode()
        }

        fun encode(data: String?): String {
            return instance.update(data!!).encode()
        }
    }
}